crates/sui-core/src/unit_tests/data/object_basics/sources/object_basics.move

```diff
@@ -105,4 +105,20 @@ module examples::object_basics {
             tx_context::sender(ctx),
         );
     }
+
+    struct Name has copy, drop, store {
+        name_str: std::string::String
+    }
+
+    public entry fun add_field_with_struct_name(o: &mut Object, v: Object) {
+        sui::dynamic_field::add(&mut o.id, Name {name_str: std::string::utf8(b"Test Name")}, v);
+    }
```

Comment:
Should we also add `dynamic_object_field` tests for this case? The process to retrieve the inner name will be a tad different because of the wrapper.

Reply:
sure I will add tests for dynamic_object_field